toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt

```diff
@@ -413,6 +400,104 @@ private fun MapViewEventHandler(
     }
 }
 
+/**
+ * Handles viewpoint change events and persistence for a [MapView].
+ *
+ * @since 200.4.0
+ */
+@Composable
+private fun ViewpointHandler(
+    mapView: MapView,
+    viewpointPersistence: ViewpointPersistence,
+    onViewpointChangedForCenterAndScale: ((Viewpoint) -> Unit)?,
+    onViewpointChangedForBoundingGeometry: ((Viewpoint) -> Unit)?,
+    onVisibleAreaChanged: ((Polygon) -> Unit)?
+) {
+    val currentOnViewpointChangedForCenterAndScale by rememberUpdatedState(
+        onViewpointChangedForCenterAndScale
+    )
+    val currentOnViewpointChangedForBoundingGeometry by rememberUpdatedState(
+        onViewpointChangedForBoundingGeometry
+    )
+    val currentVisibleAreaChanged by rememberUpdatedState(onVisibleAreaChanged)
+    val currentViewpointPersistence by rememberUpdatedState(viewpointPersistence)
+    var persistedViewpoint by rememberSaveable(
+        saver = Saver(
+            save = {
+                val viewpoint = it.value ?: return@Saver null
+
+                // If the spatial reference is pannable or geographic, normalize the viewpoint's center
+                // If we don't do this, the viewpoint will not be restored correctly after rotation
+                // if the viewpoint has crossed the central meridian. However normalization can
+                // only be done with pannable or geographic spatial references.
+                val sr = viewpoint.targetGeometry.spatialReference
+                val viewpointToPersist = if (sr?.isPannable == true || sr?.isGeographic == true) {
+                    GeometryEngine.normalizeCentralMeridian(viewpoint.targetGeometry)
+                        ?.let { normalizedGeometry ->
+                            if (viewpoint.viewpointType == ViewpointType.CenterAndScale) {
+                                Viewpoint(
+                                    center = normalizedGeometry as Point,
+                                    scale = viewpoint.targetScale,
+                                    rotation = viewpoint.rotation
+                                )
+                            } else {
+                                Viewpoint(
+                                    boundingGeometry = normalizedGeometry as Polygon,
```

Comment:
This cast should not be necessary, potentially could cause a crash:
```suggestion
                                    boundingGeometry = normalizedGeometry,
```